doc/fundamentals/updating_pipelines.md

```diff
@@ -0,0 +1,65 @@
+# Updating Pipelines
+
+During development, it's very common to update pipelines, whether it's changing your code or just cranking up parallelism.  For example, when developing a machine learning model you will likely need to try out a bunch of different versions of your model while your training data stays relatively constant.  This is where `update-pipeline` comes in.
+
+## Updating your pipeline specification
+
+In cases in which you are updating parallelism, adding another input repo, or otherwise modifying your [pipeline specification](../reference/pipeline_spec), you just need to update your JSON file and call `update-pipeline`:
+
+```sh	
+$ pachctl update-pipeline -f pipeline.json 
+```
+
+Similar to `create-pipeline`, `update-pipeline` with the `-f` flag can also take a URL if your JSON manifest is hosted on GitHub or elsewhere. 
+
+## Updating the code used in a pipeline
+
+You can also use `update-pipeline` to update the code you are using in one or more of your piplines.  To update the code in your pipeline:
+
+1. Make the code changes.
+2. Re-build you Docker image.
+3. Call `update-pipeline` with the `--push-images` flag.
+
+You need to call `update-pipeline` with the `--push-images` flag because, if you have already run your pipeline, Pachyderm has already pulled the specified images.  It won't re-pull new versions of the images, unless we tell it to (which ensures that we don't waste time pulling images when we don't need to).  When `--push-images` is specified, Pachyderm will do the following:
+
+1. Tag your image with a new unique tag.
+2. Push that tagged image to your registry (e.g., DockerHub).
+3. Update the pipeline specification that you previously gave to Pachyderm with the new unique tag.
+
+For example, you could update the Python code used in the [OpenCV pipeline](../examples/beginner_tutorial) via:
+
+```sh
+pachctl update-pipeline -f edges.json --push-images --password <registry password> -u <registry user>
+```
+
+## Archived commits
```

Comment:
archive doesnt exist in v1.4. Instead we just create a new commit-tree for the updated pipeline. We don't actually have a great name for this concept yet. @gabrielgrant and I have been just referring to them as commit-trees or pipeline versions (which is technically in the API, but isn't something we use elsewhere much afaik).